packages/flutter/lib/src/material/tabs.dart

```diff
@@ -381,19 +381,55 @@ class _TabsScrollBehavior extends BoundedBehavior {
   }
 }
 
-class TabBarSelection {
-  TabBarSelection({ int index: 0, this.maxIndex, this.onChanged }) : _index = index {
+class TabBarSelection extends StatefulComponent {
+  TabBarSelection({
+    Key key,
+    this.index,
+    this.maxIndex,
+    this.onChanged,
+    this.child
+  }) : super(key: key)  {
+    assert(child != null);
     assert(maxIndex != null);
-    assert(index != null);
-    assert(_index >= 0 && _index <= maxIndex);
+    assert((index != null) ? index >= 0 && index <= maxIndex : true);
   }
 
-  final VoidCallback onChanged;
+  final int index;
   final int maxIndex;
+  final Widget child;
+  final ValueChanged<int> onChanged;
+
+  TabBarSelectionState createState() => new TabBarSelectionState();
+
+  static TabBarSelectionState of(BuildContext context) {
+    return context.ancestorStateOfType(TabBarSelectionState);
+  }
+}
+
+abstract class TabBarSelectionPerformanceListener {
+  void handleStatusChange(PerformanceStatus status) { }
+  void handleProgressChange() { }
+  void handleDeactivate() { }
```

Comment:
This should probably be handleSelectionDeactivate(). It looks weird in the classes that implement this otherwise (handle _what_ deactivating? the tab?)